components/authz-service/engine/opa/policy/authz_v2_test.rego

```diff
@@ -350,3 +350,74 @@ test_authorized_project_matches_only_allowed_projects_when_some_projects_denied
 
 	actual_projects == {"p2"}
 }
+
+test_authorized_project_returning_single_value_with_one_project {
+    actual_projects = authorized_project with data.roles.project.actions as ["iam:teams:list"]
+    with data.policies.polid1 as {
+        "members": ["user:local:dave"],
+        "statements": {
+            "sid1": {"effect": "allow", "actions": ["*"], "resources": ["*"], "projects": ["foo-project"]},
+        },
+    }
+     with data.policies.polid2 as {
+        "members": ["user:local:dave"],
+        "statements": {
+            "sid2": {"effect": "allow", "role": "project", "resources": ["*"], "projects": ["foo-project"]},
+        },
+    }
+    with input as {"subjects": ["user:local:dave"], "action": "iam:teams:list", "resource": "iam:teams", "projects": ["foo-project", "(unassigned)"]}
+
+    actual_projects == {"foo-project"}
+}
+
+test_authorized_project_returning_single_value_with_two_projects {
+    actual_projects = authorized_project with data.roles.project.actions as ["iam:teams:list"]
```

Comment:
Notice the `authorized_project` rule here is invoked as a complete rule (no indexing).
The result can then be tested as in L389.